src/function/response.ts

```diff
@@ -9,3 +9,7 @@ export interface Response {
   body?: string
   isBase64Encoded?: boolean
 }
+
+export interface BuilderResponse extends Response {
+  ttl?: number
+}
```

Comment:
```suggestion

export interface Response extends BaseResponse {
  /**
   * The `ttl` value is only supported in `builder` functions
   */
  ttl?: never
}
export interface BuilderResponse extends BaseResponse {
  ttl?: number
}

```

I think this will work: if you rename the base interface `BaseResponse`, you can make both of them inherit, and prohibit `ttl` on `Response`

Reply:
this is great! i'll try that

Reply:
it works! thank you so much! 🙌

Reply:
actually, i think this breaks builder.ts because we wanted the wrapHandler to return a Handler and not a BuilderHandler, and a Handler uses the Response type which does not accept the ttl field: 😞 
![image](https://user-images.githubusercontent.com/10638317/142081060-abb51747-3284-43aa-af36-30f56130809b.png)

Errors:
```
Type '(event: Event, context: Context, callback: HandlerCallback<Response>) => void | BuilderResponse | Promise<{ body: string; statusCode: number; }> | Promise<...>' is not assignable to type 'Handler'.
  Type 'void | BuilderResponse | Promise<{ body: string; statusCode: number; }> | Promise<Response | BuilderResponse | { metadata: { version: number; builder_function: boolean; ttl: number; }; ... 5 more ...; isBase64Encoded?: boolean | undefined; }>' is not assignable to type 'void | Response | Promise<Response>'.
    Type 'BuilderResponse' is not assignable to type 'void | Response | Promise<Response>'.
      Type 'BuilderResponse' is not assignable to type 'Response'.
        Types of property 'ttl' are incompatible.
          Type 'number | undefined' is not assignable to type 'undefined'.
            Type 'number' is not assignable to type 'undefined'.ts(2322)
(parameter) callback: HandlerCallback<Response>
```
```
Argument of type 'BuilderResponse | { metadata: { version: number; builder_function: boolean; ttl: number; }; ttl?: number | undefined; statusCode: number; headers?: { [header: string]: string | number | boolean; } | undefined; multiValueHeaders?: { ...; } | undefined; body?: string | undefined; isBase64Encoded?: boolean | undefined; }' is not assignable to parameter of type 'Response'.
  Type 'BuilderResponse' is not assignable to type 'Response'.ts(2345)
```

Reply:
Aah, that makes sense. I'm sure we'll work something out eventually, but it's not a blocker for this.